tests/src/application/controllers/forget-password.spec.ts

```diff
@@ -95,13 +98,20 @@ describe('ForgetPassword Controller', () => {
     jest.spyOn(emailValidatorStub, 'isValid').mockImplementationOnce(() => {
       throw new Error()
     })
-    const httpRequest = {
-      body: {
-        email: 'any_email@mail.com'
-      }
-    }
-
-    const httpResponse = await sut.handle(httpRequest)
+    const httpResponse = await sut.handle(makeFakeRequest())
     expect(httpResponse).toEqual(makeFakeServerError())
   })
+
+  it('Should return 400 if no guardian is found with the provided email', async () => {
+    const { sut, forgetPasswordStub } = makeSut()
+    jest.spyOn(forgetPasswordStub, 'forgetPassword').mockReturnValueOnce(new Promise(resolve => { resolve(false) }))
```

Comment:
I think `mockReturnValueOnce(new Promise(resolve => { resolve(false) }))` is too verbose, maybe `mockResolveValueOnce(false)` is more elegant. what do you think?

Reply:
This would give the following error: "Argument of type 'boolean' is not assignable to parameter of type 'Promise<boolean>'".ts(2345)", because it is expecting a promise from a boolean and not a boolean directly

Reply:
But `mockResolveValueOnce` implicitly use a promise, different of `mockReturnValueOnce`.